File: leetcode/leetcode-94.py
from typing import List

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        ans = []
        def dfs(u):
            if u is None:
                return
            dfs(u.left)
            ans.append(u.val)
            dfs(u.right)
        dfs(root)
        return ans


if __name__ == "__main__":
    so = Solution()

File: leetcode/leetcode-94.cpp
/*
 *@File Name          :leetcode-94.cpp
 *@Author             :cc
 *@Version            :1.0.0
 *@Date               :2022/3/1 9:12
 *@Description        :
 *@Function List      :
 *@History            :
 */

#include "vector"
#include "algorithm"
#include "string"

using namespace std;


struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;

    TreeNode() : val(0), left(nullptr), right(nullptr) {}

    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}

    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<int> ans;

    void dfs(TreeNode *root) {
        if (root == nullptr) return;
        dfs(root->left);
        ans.push_back(root->val);
        dfs(root->right);
    }

    vector<int> inorderTraversal(TreeNode *root) {
        dfs(root);
        return ans;
    }
};